### PR TITLE
Kallisto fragment length params to channels, strand to meta

### PR DESCRIPTION
**Pull request by pinin4fjords**
In this PR:

- Fragment length parameters are mandatory for Kallisto in single-end, so move them to channels. 
- Put a strandedness into meta and use to determine strandedness parameters, as requested by @drpatelh , to avoid logic like [this](https://github.com/nf-core/rnaseq/blob/58bbc278ad3fd39bc0b28d63caec07ac8d83b162/conf/modules.config#L1164).
This is also consistent with the equivalent [Salmon module](https://github.com/nf-core/modules/blob/master/modules/nf-core/salmon/quant/main.nf).

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by pinin4fjords** [comment]
@nf-core-bot fix linting

File: modules/nf-core/kallisto/quant/main.nf [hunk 1]
```diff
@@ -27,12 +29,16 @@ process KALLISTO_QUANT {
     prefix = task.ext.prefix ?: "${meta.id}"
     def gtf_input = gtf ? "--gtf ${gtf}" : ''
     def chromosomes_input = chromosomes ? "--chromosomes ${chromosomes}" : ''
+    def single_end_params = meta.single_end ? "--single --fragment-length=$fragment_length --sd=$fragment_length_sd" : ''
```
Comment (SPPearce) [inline modules/nf-core/kallisto/quant/main.nf:32]: How does this behave if fragment_length and/or fragment_length_sd are empty for single ended data? Do we need to catch this case in the module code?

Comment (pinin4fjords) [reply modules/nf-core/kallisto/quant/main.nf:32]: Good point- checks added

**Review by maxulysse — APPROVED** [review]
ok for me

**Comment by pinin4fjords** [comment]
Thanks @maxulysse . @nf-core/maintainers we can argue about it later and fix consistently for both salmon and kallisto (and other modules using meta.strandedness) if that's the consensus.

**Comment by sateeshperi** [comment]
if the configs are not different now can use just as `nextflow.config`

**Pull request merged**